### PR TITLE
[Maintenance] phpstan upgrade

### PR DESCRIPTION
**Pull request by Rafikooo**
As we moved to `phpstan ^1.x` version across Sylius it would be good to do it here as well

### CODE REVIEW

File: src/Grid/Filter/ChannelFilter.php [hunk 1]
```diff
@@ -15,11 +15,18 @@
...
     {
+        Assert::isArray($data);
+
+        if (!array_key_exists('channel', $data)) {
+            return;
+        }
+
         if ($data['channel'] === '') {
             return;
         }
```
Comment (jakubtobiasz) [inline src/Grid/Filter/ChannelFilter.php:32]:
```suggestion
        if (!array_key_exists('channel', $data) || $data['channel'] === '') {
            return;
        }
```

File: src/Provider/TaxRatePercentageProvider.php [hunk 2]
```diff
@@ -34,7 +34,10 @@ public function provideFromAdjustable(AdjustableInterface $adjustable): ?string
             return null;
         }
 
-        return $this->provideFromAdjustment($taxAdjustments->first());
+        $adjustment = $taxAdjustments->first();
```
Comment (jakubtobiasz) [inline src/Provider/TaxRatePercentageProvider.php:37]: I can't mark more lines as they're out of range, but you can do something like that:

Remove the `if` with `isEmpty()` and write it like this:

```suggestion
        $adjustment = $taxAdjustments->first();
        
        if (null === $adjustment) {
            return null;
        }
        
        return $this->provideFromAdjustment($adjustment);
```

**Review by jakubtobiasz — CHANGES_REQUESTED** [review]

File: src/Provider/TaxRatePercentageProvider.php (hunk 2 above)
Comment (Rafikooo) [reply src/Provider/TaxRatePercentageProvider.php]: I think both solutions are equivalent as we need to check there is a single adjustment. The previous solution was getting this information based on collection API, now we check it directly from the returned collection element

Comment (jakubtobiasz) [reply src/Provider/TaxRatePercentageProvider.php]: I know, but after my suggestion `Assert::isInstanceOf($adjustment,AdjustmentInterface::class);` can be removed, and it should work I guess 🤔.

File: src/Doctrine/ORM/InvoiceRepository.php [hunk 3]
```diff
@@ -29,13 +30,17 @@ public function findOneByOrder(OrderInterface $order): ?InvoiceInterface
 
     public function findByOrderNumber(string $orderNumber): array
     {
-        return $this
+        $orders = $this
```
Comment (NoResponseMate) [inline src/Doctrine/ORM/InvoiceRepository.php:33]:
```suggestion
        $invoices = $this
```

File: src/Provider/TaxRatePercentageProvider.php (hunk 2 above)
Comment (jakubtobiasz) [reply src/Provider/TaxRatePercentageProvider.php]: We've solved it together 🤝🏼.

File: src/Ui/RedirectToOrderShowAction.php [hunk 4]
```diff
@@ -36,6 +36,7 @@ public function __construct(RouterInterface $router, OrderRepositoryInterface $o
     public function __invoke(Request $request): Response
     {
         $number = $request->attributes->get('number');
+        Assert::string($number);
```
Comment (NoResponseMate) [inline src/Ui/RedirectToOrderShowAction.php:39]: While you're at it, change the var name to `$orderNumber`.
Although not strictly related, a construct like `Assert::string($number)` looks icky.

File: src/Doctrine/ORM/InvoiceRepository.php [hunk 5]
```diff
@@ -29,13 +30,17 @@ public function findOneByOrder(OrderInterface $order): ?InvoiceInterface
...
             ->createQueryBuilder('invoice')
             ->innerJoin('invoice.order', 'o')
             ->where('o.number = :orderNumber')
             ->setParameter('orderNumber', $orderNumber)
             ->getQuery()
             ->getResult()
         ;
+
+        Assert::isArray($orders, 'Result of query should be an array.');
```
Comment (NoResponseMate) [inline src/Doctrine/ORM/InvoiceRepository.php:42]:
```suggestion
        Assert::isArray($invoices);
```
No point in overwriting the default message.

**Review by NoResponseMate — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, Rafał! :tada:

**Pull request merged**